services-custom/s3-transfer-manager/src/it/java/software/amazon/awssdk/transfer/s3/S3IntegrationTestBase.java

```diff
@@ -55,10 +57,16 @@
      */
     @BeforeClass
     public static void setUp() throws Exception {
+        System.setProperty("aws.crt.debugnative", "true");
         s3 = s3ClientBuilder().build();
         s3Async = s3AsyncClientBuilder().build();
     }
 
+    @AfterClass
+    public static void cleanUp() {
+        CrtResource.waitForNoResources();
```

Comment:
Nit/off-topic: Long-term, I think we should consider moving away from static resources like this, since it prohibits us from being able to run these tests in parallel one day (which is supported in JUnit 5).

Reply:
Yeah, we might be able to run some tests (eg: tests for idempotent operations) in parallel but for some service tests that require certain resources to be created first such as s3 bucket, it might be better to set up static resources to be used by all tests within the same class. If we create one bucket per test, we might run out of buckets, and it'll take more time for each test to finish because of the resource creation and deletion.